src/gt4py/analysis/passes.py

```diff
@@ -407,9 +407,9 @@ def _merge_extents(self, refs: list):
             # Merge offsets for same symbol
             for name, extent in refs:
                 if extent is None:
-                    assert name in params or name not in result
+                    assert name in params or result.get(name, Extent.zeros()) == Extent.zeros()
                     params |= {name}
-                    result.setdefault(name, Extent((0, 0), (0, 0), (0, 0)))
+                    result.setdefault(name, Extent.zeros())
```

Comment:
This is not the best way to check this, but it does the trick. If Extents were able to keep track of `None` values instead of converting these to `Extent.zeros()` it would be cleaner.

Reply:
I've spent some time looking into the bug myself and I agree the code there is not clean. I think it would be better to simplify it and replace the whole contents of the loop by:

```
                extent = extent or Extent.zeros()
                result.setdefault(name, Extent.zeros())
                result[name] |= extent
```
and also deleting the unnecessary `params` variable.